### PR TITLE
add addon to apis

### PR DESCRIPTION
**Pull request by haoqing0110**
@haoqing0110 for a change like this, we definitely need an issue because it takes time to edit this...we also need to see the TRAIN label on this issue so we can relate it to the train and prioritize it as such. Thank you so much!

Related Jira https://issues.redhat.com/browse/ACM-5055

### CODE REVIEW

**Comment by haoqing0110** [comment]
Thanks @dockerymick and cc @cadawson , plz hold this PR and we are waiting for response in https://issues.redhat.com/browse/ACM-2200 whether we are going to release this feature in 2.8.

File: apis/addondeploymentconfig.json.adoc [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+[[_rhacm-docs_apis_addondeploymentconfig_jsonaddondeploymentconfig_nodePlacement]]
+*nodePlacement*
+
+[options="header", cols=".^2a,.^3a,.^4a"]
+|===
+|Name|Description|Schema
+|*nodeSelector* +
+__optional__|Defines which nodes the pods are scheduled to be enabled. If the selector is an empty list, the selector matches all nodes.|map[string]string
+|*tolerations* +
+__optional__|Connected by pods and used to list any taint that matches the triple <key,value,effect> using the matching operator <operator>.|[]corev1.Toleration
```
Comment (dockerymick) [inline apis/addondeploymentconfig.json.adoc]: @haoqing0110 I made an update here, can you confirm that my change is accurate

Comment (haoqing0110) [reply apis/addondeploymentconfig.json.adoc]: @dockerymick It seems to have changed a little bit of meaning. 
For the node selector, it defines which Nodes the Pods will be scheduled and running on, using "enabled" is not accurate. 
For the tolerations, it is to tolerate any taint that matches the triple <key,value,effect> using the matching operator <operator>.

Comment (dockerymick) [reply apis/addondeploymentconfig.json.adoc]: "For the node selector, it defines which Nodes the Pods will be scheduled and **running on**" seems incomplete. where do the pods and nodes run?  Is it running on the hub cluster? managed cluster? all clusters? 

Pod exceptions for any taint that matches the triple <key,value,effect> using the matching operator <operator>.|[]corev1.Toleration

Comment (haoqing0110) [reply apis/addondeploymentconfig.json.adoc]: The pod will be running on the nodes. 

The nodeSelector defined in this API will finally be applied to pods. The meaning of node selector is the same as k8s pods' [nodeselector](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector). 

For the toleration, the toleration defined in this API will also finally be applied to pods, and the meaning of toleration is the same as k8s [taint and toleration](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/). 

Hope the above info can help better understand this field.

Comment (dockerymick) [reply apis/addondeploymentconfig.json.adoc]: @haoqing0110 thanks for the sources, I misunderstood at first. Making updates now

Comment (dockerymick) [reply apis/addondeploymentconfig.json.adoc]: @haoqing0110 I updated `nodeSelector` to read: Define which nodes the pods are scheduled to run on. If the selector has an empty list, the selector matches all nodes.

When the selector matches all nodes, is that done automatically and how is it matched? When you say selector, are we referring to the user or to the parameter name (`nodeSelector`). Based on the reference that you provided, the pods are scheduled to nodes that have `nodeSelector` labels.

for tolerations: Applied to pods and used to schedule pods to any taint that matches the `<key,value,effect>` toleration using the matching operator (`<operator>`).

Comment (haoqing0110) [reply apis/addondeploymentconfig.json.adoc]: > When the selector matches all nodes, is that done automatically and how is it matched? 

selector refer to the `nodeSelector`. when the `nodeSelector` is empty, so no change will make to the pods, it's by default selecting all the nodes.

Comment (haoqing0110) [reply apis/addondeploymentconfig.json.adoc]: @dockerymick The update looks good to me. Thx!

File: apis/clustermanagementaddon.json.adoc [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+[[_rhacm-docs_apis_clustermanagementaddon_jsonqueryclustermanagementaddon]]
+=== Query a single ClusterManagementAddOn
+....
+GET /addon.open-cluster-management.io/v1alpha1/clustermanagementaddons/{clustermanagementaddon_name}
+....
+
+
+==== Description
+Query a single ClusterManagementAddOn for more details.
```
Comment (oafischer) [inline apis/clustermanagementaddon.json.adoc:158]: Is the "for more details" supposed to be there?

Comment (dockerymick) [reply apis/clustermanagementaddon.json.adoc:158]: Yea, the user will receive more details about the add-on. @haoqing0110 will you confirm?

File: apis/managedclusteraddon.json.adoc [hunk 3]
```diff
@@ -0,0 +1,274 @@
+[#managedclusteraddon-api]
+= ManagedClusterAddOn API (v1alpha1)
+
+[[_rhacm-docs_apis_managedclusteraddon_jsonoverview]]
+== Overview
+This documentation is for the ManagedClusterAddOn resource for {product-title}. The ManagedClusterAddOn resource has four possible requests: create, query, delete, and update.
```
Comment (oafischer) [inline apis/managedclusteraddon.json.adoc:6]: We should avoid saying "this documentation". Maybe something like "see the following overview for..." instead.

File: apis/clustermanagementaddon.json.adoc [hunk 4]
```diff
@@ -0,0 +1,299 @@
+[#clustermanagementaddon-api]
+= ClusterManagementAddOn API (v1alpha1)
+
+[[_rhacm-docs_apis_clustermanagementaddon_jsonoverview]]
+== Overview
+This documentation is for the ClusterManagementAddOn resource for {product-title}. The ClusterManagementAddOn resource has four possible requests: create, query, delete, and update.
```
Comment (oafischer) [inline apis/clustermanagementaddon.json.adoc:6]: Same here regarding "this documentation".

File: apis/addondeploymentconfig.json.adoc [hunk 5]
```diff
@@ -0,0 +1,285 @@
+[#addondeploymentconfig-api]
+= AddOnDeploymentConfig API (v1alpha1)
+
+[[_rhacm-docs_apis_addondeploymentconfig_jsonoverview]]
+== Overview
+This documentation is for the AddOnDeploymentConfig resource for {product-title}. The AddOnDeploymentConfig resource has four possible requests: create, query, delete, and update.
```
Comment (oafischer) [inline apis/addondeploymentconfig.json.adoc:6]: Same here regarding "this documentation".

**Review by oafischer — APPROVED** [review]
One suggestion but otherwise looks good, thanks!

File: apis/clustermanagementaddon.json.adoc (hunk 2 above)
Comment (haoqing0110) [reply apis/clustermanagementaddon.json.adoc:158]: yes, correct.

File: apis/managedclusteraddon.json.adoc (hunk 3 above)
Comment (haoqing0110) [reply apis/managedclusteraddon.json.adoc:6]: "This documentation" is referring to other API docs in https://github.com/stolostron/rhacm-docs/tree/2.8_stage/apis, to keep consistency, personally not recommended to modify it here. Maybe another PR to modify the expression.

Comment (dockerymick) [reply apis/managedclusteraddon.json.adoc:6]: @haoqing0110 I agree with your suggestion. @oafischer I also agree with yours. Therefore, let's create a separate do issue to update that line in all of the PRs. Could be a practice issue for Faisa, I'll leave that decision for you Oliver. Let me know what your thoughts

**Comment by dockerymick** [comment]
@oafischer thanks for your review and feedback. I agree with your suggestions. As discussed in Slack, we will make the statement updates in a different issue and PR. Merging this

**Pull request merged**